program/Project Descriptions and Plans/CORE-V-HAL/Project-Concept-HAL.md

```diff
@@ -0,0 +1,155 @@
+
+It will be a key component of a Core-V MCU SDK.
+
+ 
+## Initial Estimate of Timeline
+The project will be last for 3 Months and have below output at different stage.
+
+- To agree the key technical requirements, approach, and specification for a Core-V HAL that will be re-used for all Core-V Cores
+
+Once the HAL approach is agreed
+   - To implement basic peripherals (UART, TIME, INT...) on Core-V MCU.
+   - To have FreeRTOS run  on Core-V MCU in conjunction with HAL SDK support 
+   - To implement support other peripherals and meet application requirement.
+
+## Explanation of why OpenHW should do this project
+
+- The HAL is needed to establish a baseline approach for addressing the Core-V Cores and peripherals under a consistent approach. It is a cornerstone of all Core-V SDK and software work.
+
+- There are some challenges for Core-V MCU users to develop application bases on adopted PULP SDK due to the complexity of the development configuration environment. The PULP SDK is not viewed as a basis for future development work. 
+
+- There should be an out-of-box SDK solution offered by OpenHW supporting Core-V MCU platform developer.
+
+- An easy-to-use SDK enhances the Core-V MCU development environment by reducing the barrier on software development and enabling to create reusable application code.
+
+- There are some requirement from system tools (such as compiler) to help verify their design's performance and correctness using HAL SDK.
+
+## Industry landscape: description of competing, alternative, or related efforts in the industry
+There are also some similar standards in market currently, the most representative HAL is below:
+   1. CMSIS for ARM cortex A and M architecture
+   2. NMSIS for RISC-V architecture
+   3. CSI for RISC-V architecture
+   4. FreeRTOS CommonIO specification https://developer.amazon.com/acs-devices
+
+### Table showing how existing solutions map against requirements at Project Concept gate
+
+| Items | CMSIS | NMSIS | CSI | CommonIO | 
+| --- | --- | --- | --- | --- |
+| Core | Support | Support | Support | Support |
+| Driver | Device x13 | - | Device x 23 | |
+| RTOS | FreeRTOS | - | FreeRTOS, Rhino | |
+| NN | Support | Support | Support | |
+| DSP | Support | Support | Support | |
+| Coding Rule | MISRA | MISRA | MISRA, TUeV61508 | |
+| Validation | Support | Support | Support | |
+| License | Apache 2.0 | Apache 2.0 | Apache 2.0 | |
+| Software pack | Support | Support | Support | |
+| **Numbered Requirements**| Y/N/Maybe |  |  | |
+| T1 |  |  |  |  |
+| T2 |  |  |  |  |
+| T3 |  |  |  |  |
+| T4 |  |  |  |  |
+| T5 |  |  |  |  |
+| T6 |  |  |  |  |
+| T7 |  |  |  |  |
+| T8 |  |  |  |  |
+| T9 |  |  |  |  |
+| IA1 |  |  |  |  |
+| S1 |  |  |  |  |
+| S2 |  |  |  |  |
+| S3 |  |  |  |  |
+| S4 |  |  |  |  |
+| IM1 |  |  |  |  |
+| IM2 |  |  |  |  |
+| IM3 |  |  |  |  |
+| IM4 |  |  |  |  |
+
```

Comment:
| Category     	  		| Req #		| CMSIS | NMSIS | CSI | Requirement descriptions                 |
|----------	  		    |------		| ----- | ----- |---- | ------------------------	|
| *Technical Reqmtns*	| T-1     	|  N    | Y     | Y   |HAL MUST support API for core features | 
| 			  	        | T-2     	|  Y    | N     | Y   |HAL MUST support API for on-chip peripheral features | 
| 			  	        | T-3     	|  Y    | Y     | Y   |HAL MUST support both RTOS and bare-metal scenarios |
| 			  	        | T-4     	|  Y    | Y     | Y   |HAL MUST be able to support bare metal libc |
| 			  	        | T-5     	|  Y    | Y     | Y   |HAL MUST be able to be extended to support all CORE-V cores for the purpose of RTOS (not Linux) support |
| 			  	        | T-6     	|  Y    | N     | N   |HAL MUST support a description of the chip using a standardized syntax |
| 			  	        | T-7     	|  Y    | N     | N   |HAL SHOULD define a configuration file that can indicate to the IDE low level chip parameters such as memory space and allow it to display/read/write the registers of peripherals |
| 			  	        | T-8     	| maybe | maybe |maybe|HAL MAY be able to be extended to support the CORE-V X Coprocessor Interface (CV-X-IF) |
| 			  	        | T-9     	| maybe | maybe |maybe|HAL MAY be able to be extended to support the CV-VEC coprocessor | 
| *Industry Adoption*	| IA-1     	|  N    | maybe | Y   |HAL SHOULD be adopted/implemented for other RISC-V devices | 	
| *Specification Rqmtns*| S-1     	|  Y    | Y     | Y   |HAL Specification License for CORE-V MUST be open-source or freely licensed | 			  	| T-2     		| 
| 				        | S-2     	|  Y    | Y     | Y   |Proprietary extensions to the HAL Specification and Implementation, such as for custom ISA extensions or proprietary chip, MUST be supportable/allowable under open-source license|
| 				        | S-3     	|  Y    | Y     | Y   |HAL Specification, user documentation, and system implementation documentation for CORE-V HAL MUST be available in English |
| 				        | S-4     	|  N    | N     | N   |HAL Specification MUST be published and managed by an industry open source or standards body, such as OpenHW, RISC-V International, or FOSSi |
| *Implementation Rqmtns*| IM-1    	|  -    | -     | -   |A complete open source reference implementation of the HAL for CORE-V MUST be a project deliverable |
| 				        | IM-2     	|  -    | -     | -   |FreeRTOS SHOULD be supported in the open source reference implementation of HAL for CORE-V |
| 				        | IM-3     	|  -    | -     | -   |Newlib SHOULD be supported in the open source reference implementation  of HAL for CORE-V |
| 				        | IM-3     	|  -    | -     | -   |The HAL reference implementation for CORE-V SHOULD be published by the body publishing the Specification (previous page) |

Reply:
@shangyunhai Nice work. You can leave out the Req # column, since it was only used to refer to the earlier table, and it can be removed from that earlier table as well.  We should have a column for Amazon's FreeRTOS HAL (perhaps just reuse  the Req # column), which can be just filled with "To Be Done" (TBD).  We don't need the information for Project Concept, just to show that it is under consideration.